packages/cli/src/commands/hydrogen/build-vite.ts

```diff
@@ -0,0 +1,233 @@
+      env: 'SHOPIFY_HYDROGEN_FLAG_SOURCEMAP',
+      allowNo: true,
+      default: true,
+    }),
+    'lockfile-check': Flags.boolean({
+      description:
+        'Checks that there is exactly 1 valid lockfile in the project.',
+      env: 'SHOPIFY_HYDROGEN_FLAG_LOCKFILE_CHECK',
+      default: true,
+      allowNo: true,
+    }),
+    'disable-route-warning': Flags.boolean({
+      description: 'Disable warning about missing standard routes.',
+      env: 'SHOPIFY_HYDROGEN_FLAG_DISABLE_ROUTE_WARNING',
+    }),
+    codegen: commonFlags.codegen,
+    'codegen-config-path': commonFlags.codegenConfigPath,
+    diff: commonFlags.diff,
+  };
+
+  async run(): Promise<void> {
+    const {flags} = await this.parse(Build);
+    const originalDirectory = flags.path
+      ? resolvePath(flags.path)
+      : process.cwd();
+    let directory = originalDirectory;
+
+    if (flags.diff) {
+      directory = await prepareDiffDirectory(originalDirectory, false);
+    }
+
+    await runViteBuild({
+      ...flagsToCamelObject(flags),
+      useCodegen: flags.codegen,
+      directory,
+    });
+
+    if (flags.diff) {
+      await copyDiffBuild(directory, originalDirectory);
+    }
+
+    // The Remix compiler hangs due to a bug in ESBuild:
+    // https://github.com/evanw/esbuild/issues/2727
+    // The actual build has already finished so we can kill the process.
+    process.exit(0);
+  }
+}
+
+type RunBuildOptions = {
+  entry?: string;
+  directory?: string;
+  useCodegen?: boolean;
+  codegenConfigPath?: string;
+  sourcemap?: boolean;
+  disableRouteWarning?: boolean;
+  assetPath?: string;
+  bundleStats?: boolean;
+  lockfileCheck?: boolean;
+};
+
+export async function runViteBuild({
+  entry: ssrEntry,
+  directory,
+  useCodegen = false,
+  codegenConfigPath,
+  sourcemap = false,
+  disableRouteWarning = false,
+  lockfileCheck = true,
+  assetPath = '/',
+}: RunBuildOptions) {
+  if (!process.env.NODE_ENV) {
+    process.env.NODE_ENV = 'production';
+  }
+
+  const root = directory ?? process.cwd();
+
+  if (lockfileCheck) {
+    await checkLockfileStatus(root, isCI());
+  }
+
+  console.time(LOG_WORKER_BUILT);
+
+  outputInfo(`\n🏗️  Building in ${process.env.NODE_ENV} mode...\n`);
+
+  const {
+    userViteConfig,
+    remixConfig,
+    clientOutDir,
+    serverOutDir,
+    serverOutFile,
+  } = await getViteConfig(root);
+
+  const serverMinify = userViteConfig.build?.minify ?? true;
+  const commonConfig = {
+    root,
+    mode: process.env.NODE_ENV,
+    base: assetPath,
+  };
+
+  // Avoid static imports because this file is imported by `deploy` command,
+  // which must have a hard dependency on 'vite'.
+  const vite = await import('vite');
+
+  // Client build first
+  await vite.build({
+    ...commonConfig,
+    build: {
+      emptyOutDir: true,
+      copyPublicDir: true,
+      // Disable client sourcemaps in production
+      sourcemap: process.env.NODE_ENV !== 'production' && sourcemap,
```

Comment:
I assumed that's the case since the problem I think is in mixing front and back end code in the same file, not the sourcemap tool itself. But will double check.

Reply:
So this is Remix showing a warning on `sourcemap: true`:

<img width="382" alt="image" src="https://github.com/Shopify/hydrogen/assets/1634092/0251a324-5f05-4d65-888b-57b591868bdf">